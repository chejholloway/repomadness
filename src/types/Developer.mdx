## Developer Interface

Represents a GitHub developer's information.

This interface defines the structure of information associated with a GitHub developer.
It includes details such as the developer's username, profile URL, avatar URL, and more.

### Interface Properties

- `avatar_url`: string
- `bio`: string
- `blog`: string
- `company`: string
- `created_at`: string
- `email`: any
- `events_url`: string
- `followers_url`: string
- `followers`: number
- `following_url`: string
- `following`: number
- `gists_url`: string
- `gravatar_id`: string
- `hireable`: any
- `html_url`: string
- `id`: number
- `location`: string
- `login`: string
- `name`: string
- `node_id`: string
- `organizations_url`: string
- `public_gists`: number
- `public_repos`: number
- `received_events_url`: string
- `repos_url`: string
- `site_admin`: boolean
- `starred_url`: string
- `subscriptions_url`: string
- `twitter_username`: string
- `type`: string
- `updated_at`: string
- `url`: string

### Example Usage

```typescript
const developer: Developer = {
  avatar_url: "https://example.com/avatar.jpg",
  bio: "Software developer passionate about...",
  // ...other properties
};
