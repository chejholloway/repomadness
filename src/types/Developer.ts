export default interface Developer {
  login: string;
  avatar_url: string;
  twitter_username: string;
  // Add more properties as needed
  // Example: avatar_url: string;
}
