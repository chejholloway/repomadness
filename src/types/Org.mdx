## Org Interface

Represents a GitHub organization's information.

This interface defines the structure of information associated with a GitHub organization.
It includes details such as the organization's name, profile URL, avatar URL, and more.

### Interface Example

```ts
const organization: Org = {
  login: "exampleOrg",
  id: 789,
  avatar_url: "https://example.com/org-avatar.jpg",
  // ...other properties
};
