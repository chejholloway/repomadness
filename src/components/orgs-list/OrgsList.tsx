import React from "react";
import Org from "../../types/Org";
import OrgCard from "../cards/OrgCard";
import GenericList from "../generic-list/GenericList";
import Developer from "../../types/Developer";

interface OrgsListProps {
  developer: Developer | undefined;
  org: Org[] | undefined;
}

const OrgsList: React.FC<OrgsListProps> = ({ org }) => {
  const mapToLink = (org: Org) =>
    `${process.env.REACT_APP_GITHUB_URL}${org.login}`;
  const renderItem = (org: Org) => <OrgCard org={org} />;
  const renderCard = (org: Org) => <OrgCard org={org} />;

  return (
    <GenericList
      title="Organizations"
      data={org}
      renderItem={renderItem}
      emptyMessage="No organizations available."
      mapToLink={mapToLink}
      renderCard={renderCard}
    />
  );
};

export default OrgsList;
