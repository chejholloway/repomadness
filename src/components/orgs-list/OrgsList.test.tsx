import React from "react";
import { render } from "@testing-library/react";
import Org from "../../types/Org";
import Developer from "../../types/Developer";
import OrgsList from "./OrgsList";

jest.mock("react-router-dom");
jest.mock("@material-tailwind/react");
jest.mock("../../types/Org");
jest.mock("../../types/Developer");

const mockOrgs: Org[] = [
  {
    avatar_url: "https://avatars.githubusercontent.com/u/28929274?v=4",
    description: "Plataforma de educaÃ§Ã£o em tecnologia ðŸš€",
    events_url: "https://api.github.com/users/charlenopires/events{/privacy}",
    hooks_url: "https://api.github.com/orgs/Rocketseat/hooks",
    id: 28929273,
    issues_url: "https://api.github.com/orgs/Rocketseat/issues",
    login: "Rocketseat",
    members_url: "https://api.github.com/orgs/Rocketseat/members{/member}",
    node_id: "MDEyOk9yZ2FuaXphdGlvbjI4OTI5Mjc0",
    public_members_url:
      "https://api.github.com/orgs/Rocketseat/public_members{/member}",
    repos_url: "https://api.github.com/orgs/Rocketseat/repos",
    url: "https://api.github.com/orgs/Rocketseat",
  },
  {
    avatar_url: "https://avatars.githubusercontent.com/u/28929274?v=4",
    description: "Plataforma de educaÃ§Ã£o em tecnologia ðŸš€",
    events_url: "https://api.github.com/users/charlenopires/events{/privacy}",
    hooks_url: "https://api.github.com/orgs/Rocketseat/hooks",
    id: 28929274,
    issues_url: "https://api.github.com/orgs/Rocketseat/issues",
    login: "Rocketseat",
    members_url: "https://api.github.com/orgs/Rocketseat/members{/member}",
    node_id: "MDEyOk9yZ2FuaXphdGlvbjI4OTI5Mjc0",
    public_members_url:
      "https://api.github.com/orgs/Rocketseat/public_members{/member}",
    repos_url: "https://api.github.com/orgs/Rocketseat/repos",
    url: "https://api.github.com/orgs/Rocketseat",
  },
];

const mockDeveloper: Developer = {
  login: "ruanyf",
  avatar_url: "https://api.github.com/users/ruanyf/starred{/owner}{/repo}",
  twitter_username: "@testser",
};

describe("<OrgsList>", () => {
  it("should render component", () => {
    const { container } = render(
      <OrgsList developer={mockDeveloper} org={mockOrgs} />,
    );

    expect(container).toMatchSnapshot();
  });
});
