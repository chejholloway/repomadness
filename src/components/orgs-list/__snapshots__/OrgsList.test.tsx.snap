// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrgsList> should render component 1`] = `
<div>
  <div
    class="text-gray-700"
  >
    <h1
      class="font-normal text-center css-qhut37"
    >
      Organizations
    </h1>
    <div />
    <div />
  </div>
</div>
`;
